### DISCUSSION TITLE
User handling

### DISCUSSION THREAD

**Discussion by csbence10** [rrousselGit/riverpod#2621] [Q&A]
Hey!

How do you guys handling user sessions with Riverpod? (using auto_route for navigation)

I'm having issues finding a proper way to handle user sessions. I have an async notifier fetching the user when the correct credentials passed. I have other providers depending on this one and I want to keep them alive until I have a logged in user.

I tried to write a `ref` extension to listen to changes in the user provider but it didn't work because it always had value, therefore the providers I wanted to keep alive kept alive and when I logged out the user provider didn't dispose.

Previously, before the code generation I had a `StateNotifier` and updated the user by hand but I want to implement it in a more declarative way.

**Comment by rignaneseleo** [comment]
any solution to this?
